frontend/src/lib/utils.tsx

```diff
@@ -1079,3 +1079,7 @@ export function validateJsonFormItem(_: any, value: string): Promise<string | vo
         return Promise.reject('Not valid JSON!')
     }
 }
+
+export function ensureStringIsNotBlank(s?: string | null): string | null {
+    return (s ?? '').trim() === '' ? null : s ?? null
+}
```

Comment:
Nit/creep: 1) tests? 2) Something like `typeof s === 'string' && s.trim() !== '' ? s : null` seems cleaner and safer

Reply:
1) ah yes, absolutely 2) changing this too